OIDC code flow tests failing in native on CI

The OIDC code flow tests are now consistently failing on CI with something like:

```
Caused by: org.opentest4j.AssertionFailedError: expected: </index.html> but was: <null>
	at org.junit.jupiter.api.AssertionUtils.fail(AssertionUtils.java:55)
	at org.junit.jupiter.api.AssertionUtils.failNotEqual(AssertionUtils.java:62)
	at org.junit.jupiter.api.AssertEquals.assertEquals(AssertEquals.java:182)
	at org.junit.jupiter.api.AssertEquals.assertEquals(AssertEquals.java:177)
	at org.junit.jupiter.api.Assertions.assertEquals(Assertions.java:1124)
	at io.quarkus.it.keycloak.CodeFlowTest.testAccessTokenInjection(CodeFlowTest.java:164)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at io.quarkus.test.junit.QuarkusTestExtension.runExtensionMethod(QuarkusTestExtension.java:360)
	... 62 more

2020-02-06 17:27:14,690 INFO  [io.quarkus] (main) Quarkus stopped in 0.069s

[INFO] 
[INFO] Results:
[INFO] 
[ERROR] Errors: 
[ERROR]   CodeFlowTest.testAccessAndRefreshTokenInjection » Runtime org.opentest4j.Asser...
[ERROR]   CodeFlowTest.testAccessAndRefreshTokenInjectionWithoutIndexHtml » Runtime org....
[ERROR]   CodeFlowTest.testAccessTokenInjection » Runtime org.opentest4j.AssertionFailed...
[ERROR]   CodeFlowTest.testCodeFlowNoConsent » Runtime org.opentest4j.AssertionFailedErr...
[ERROR]   CodeFlowTest.testIdTokenInjection » Runtime org.opentest4j.AssertionFailedErro...
[ERROR]   CodeFlowTest.testIdTokenInjectionWithoutRestoredPath » Runtime org.opentest4j....
[ERROR]   CodeFlowTest.testTokenTimeoutLogout » Runtime org.opentest4j.AssertionFailedEr...
```

/cc @sberyozkin @pedroigor

--- a/integration-tests/oidc-code-flow/src/main/java/io/quarkus/it/keycloak/CustomTenantResolver.java
+++ b/integration-tests/oidc-code-flow/src/main/java/io/quarkus/it/keycloak/CustomTenantResolver.java
@@ -1,23 +1,15 @@
 package io.quarkus.it.keycloak;
 
-import java.util.List;
+import javax.enterprise.context.ApplicationScoped;
 
-// import javax.enterprise.context.ApplicationScoped;
 import io.quarkus.oidc.TenantResolver;
-import io.vertx.core.http.Cookie;
 import io.vertx.ext.web.RoutingContext;
 
-//@ApplicationScoped
+@ApplicationScoped
 public class CustomTenantResolver implements TenantResolver {
 
     @Override
     public String resolve(RoutingContext context) {
-        List<String> tenantidList = context.queryParam("tenantid");
-        if (tenantidList.isEmpty()) {
-            Cookie cookie = context.getCookie("tenantid");
-            return cookie != null ? cookie.getValue() : null;
-        }
-        context.response().addCookie(Cookie.cookie("tenantid", tenantidList.get(0)));
-        return tenantidList.get(0);
+        return context.request().path().contains("callback-") ? "tenant-1" : null;
     }
 }

--- a/integration-tests/oidc-code-flow/src/main/java/io/quarkus/it/keycloak/ProtectedResource.java
+++ b/integration-tests/oidc-code-flow/src/main/java/io/quarkus/it/keycloak/ProtectedResource.java
@@ -2,6 +2,7 @@ package io.quarkus.it.keycloak;
 
 import javax.inject.Inject;
 import javax.ws.rs.GET;
+import javax.ws.rs.InternalServerErrorException;
 import javax.ws.rs.Path;
 
 import org.eclipse.microprofile.jwt.JsonWebToken;
@@ -42,8 +43,14 @@ public class ProtectedResource {
     }
 
     @GET
-    @Path("callback")
-    public String getNameCallback() {
+    @Path("callback-before-redirect")
+    public String getNameCallbackBeforeRedirect() {
+        throw new InternalServerErrorException("This method must not be invoked");
+    }
+
+    @GET
+    @Path("callback-after-redirect")
+    public String getNameCallbackAfterRedirect() {
         return "callback:" + getName();
     }
 

--- a/integration-tests/oidc-code-flow/src/main/resources/application.properties
+++ b/integration-tests/oidc-code-flow/src/main/resources/application.properties
@@ -10,12 +10,12 @@ quarkus.http.cors=true
 quarkus.oidc.application-type=web-app
 
 # Tenant which does not need to restore a request path after redirect
-#quarkus.oidc.tenant-1.auth-server-url=${keycloak.url}/realms/quarkus
-#quarkus.oidc.tenant-1.client-id=quarkus-app
-#quarkus.oidc.tenant-1.credentials.secret=secret
-#quarkus.oidc.tenant-1.token.issuer=${keycloak.url}/realms/quarkus
-#quarkus.oidc.tenant-1.authentication.redirect-path=/web-app/callback
-#quarkus.oidc.tenant-1.authentication.restore-path-after-redirect=false
+quarkus.oidc.tenant-1.auth-server-url=${keycloak.url}/realms/quarkus
+quarkus.oidc.tenant-1.client-id=quarkus-app
+quarkus.oidc.tenant-1.credentials.secret=secret
+quarkus.oidc.tenant-1.token.issuer=${keycloak.url}/realms/quarkus
+quarkus.oidc.tenant-1.authentication.redirect-path=/web-app/callback-after-redirect
+quarkus.oidc.tenant-1.authentication.restore-path-after-redirect=false
 
 quarkus.http.auth.permission.roles1.paths=/index.html
 quarkus.http.auth.permission.roles1.policy=authenticated

--- a/integration-tests/oidc-code-flow/src/test/java/io/quarkus/it/keycloak/CodeFlowTest.java
+++ b/integration-tests/oidc-code-flow/src/test/java/io/quarkus/it/keycloak/CodeFlowTest.java
@@ -9,7 +9,6 @@ import java.io.IOException;
 import java.net.URI;
 
 import org.hamcrest.Matchers;
-import org.junit.jupiter.api.Disabled;
 import org.junit.jupiter.api.Test;
 
 import com.gargoylesoftware.htmlunit.SilentCssErrorHandler;
@@ -131,10 +130,9 @@ public class CodeFlowTest {
     }
 
     @Test
-    @Disabled
     public void testIdTokenInjectionWithoutRestoredPath() throws IOException, InterruptedException {
         try (final WebClient webClient = createWebClient()) {
-            HtmlPage page = webClient.getPage("http://localhost:8081/web-app?tenantid=tenant-1");
+            HtmlPage page = webClient.getPage("http://localhost:8081/web-app/callback-before-redirect");
             assertNotNull(getStateCookieStateParam(webClient));
             assertNull(getStateCookieSavedPath(webClient));
 
@@ -148,7 +146,6 @@ public class CodeFlowTest {
             page = loginForm.getInputByName("login").click();
 
             assertEquals("callback:alice", page.getBody().asText());
-            // Clear the 'tokenid' cookie
             webClient.getCookieManager().clearCookies();
             // The same code path which successfully clears the cookie for all the other tests is not effective here.
             // The most likely reason is that the state cookie was created in response to "http://localhost:8081/web-app"

